src/Static.jl

```diff
@@ -939,4 +930,32 @@ function Base.show(io::IO, m::MIME"text/plain", @nospecialize(x::NDIndex))
     show(io, m, Tuple(x))
 end
 
+# field and property accessors
+"""
+    field_type(::Type{T}, f)
+
+Functionally equivalent to `fieldtype(T, f)` except `f` may be a static type.
+"""
+@inline field_type(T::Type, f::Union{Int, Symbol}) = fieldtype(T, f)
+@inline field_type(::Type{T}, ::StaticInt{N}) where {T, N} = fieldtype(T, N)
+@inline field_type(::Type{T}, ::StaticSymbol{S}) where {T, S} = fieldtype(T, S)
+
+(::Base.Fix2{typeof(getfield),<:Union{StaticSymbol{f},StaticInt{f}}})(x) where {f} = getfield(x, f)
+(::Base.Fix2{typeof(fieldtype),<:Union{StaticSymbol{f},StaticInt{f}}})(x) where {f} = fieldtype(x, f)
+
+Base.getproperty(x, ::StaticSymbol{S}) where {S} = getproperty(x, S)
+Base.setproperty!(x, ::StaticSymbol{S}, v) where {S} = setproperty!(x, S, v)
+Base.hasproperty(x, ::StaticSymbol{S}) where {S} = hasproperty(x, S)
+
+Base.getindex(nt::NamedTuple, ::StaticSymbol{S}) where {S} = getfield(nt, S)
+function Base.getindex(nt::NamedTuple, idxs::Tuple{<:StaticSymbol,Vararg{<:StaticSymbol}})
+    NamedTuple{known(idxs)}(nt)
+end
+function Base.setindex(nt::NamedTuple, v, ::StaticSymbol{S}) where {S}
+    merge(nt, NamedTuple{(S,)}((v,)))
+end
+function Base.setindex(nt::NamedTuple, vs, idxs::Tuple{Vararg{<:StaticSymbol}})
```

Comment:
Shouldn't this be
```suggestion
function Base.setindex(nt::NamedTuple, vs, idxs::Tuple{<:StaticSymbol,Vararg{<:StaticSymbol}})
```
as above, to avoid defining a method for `Tuple{}`?